docusaurus/docs/iOS/swiftui/components/list-tap-events.md

```diff
@@ -31,9 +31,9 @@ class CustomFactory: ViewFactory {
     
     public static let shared = CustomFactory()
     
-    func makeChannelDestination() -> (ChatChannel) -> CustomChannelDestination {
-        { channel in
-            CustomChannelDestination(channel: channel)
+    func makeChannelDestination() -> (ChannelSelectionInfo) -> CustomChannelDestination {
+        { selectionInfo in
+            CustomChannelDestination(channel: selectionInfo.channel)
```

Comment:
This actually surprising in my view, it feels the older a approach made more sense. You would give the Channel Info, and get back a destination, why only the selection matters now? 🤔 This is just out of curiosity

Reply:
We need an optional message too. Basically, when you perform search on messages from the channel list, they are pairs of (channel, message). When you select a search result, you should scroll up to the message in the message list. It will also be needed for deeplinking.

Reply:
Then, wouldn't it make more sense to have `ChannelInfo` where it contains the `channel` and `selectedMessage`? From what I understood the selection info is actually from the message and not the channel itself?

Reply:
It's already like that. The selection info comes from the channel, and the message is optional parameter that, if present, makes the message list scroll to the provided message.

Reply:
Ah I see... I understood it now